### PR TITLE
Code review cleanup

### PR DESCRIPTION
**Pull request by thibaudcolas**
First half of my code review – I thought it would be simpler for me to make at least some of the changes directly rather than report them in an issue.

Here are the biggest changes:

- Added a continuous integration setup with GitHub Actions
- A lot of reformatting because there were conflicting tab width definitions between the `.editorconfig` and the Prettier configuration (the PR will be much simpler to review with [?w=1](https://github.com/torchbox/tbxforms/pull/5/files?w=1).
- Fixed the JS linting setup
- Fixed autoprefixing for Safari, and noted the needed polyfills for this to work
- Switched font size definitions to using rem.

### CODE REVIEW

File: .browserslistrc
Comment (thibaudcolas) [inline .browserslistrc:5]: Without this, we’d only target Safari 15 and up.

File: .prettierrc.toml
Comment (thibaudcolas) [inline .prettierrc.toml:9]: We want to be using the definitions from `.editorconfig`, which are different between file types.

**Comment by thibaudcolas** [comment]
Based on this first pass, here is what I’m planning to further review/test:

## IE11

- [ ] Events firing IE11

## DOM APIs

- [ ] conditional field clearing for textarea (setting `.value` rather than `.innerHTML`)
- [ ] conditional field clearing for input type submit / reset (setting `.value` which is used as a label)
- [ ] conditional field clearing for select (`selectedIndex`)
- [ ] conditional field clearing for radio and checkbox (`.value`)

## Security

- [ ] Conditional fields implementation using `literal_eval` (security risk?)
- [ ] No escaping for field labels, help text, attributes, values (XSS vector)

## Accessibility

- [ ] `aria-describedby` implementation (help text + errors + char count)
- [ ] Live regions implementation
- [ ] `aria-invalid="true"` not in use?

**Comment by kbayliss** [comment]
@thibaudcolas awesome, this is super clear and concise, thanks so much - I really appreciate it.

Do you have an idea when you'll be able to continue the review? Unsure if I should approve this MR now, or wait until you've had a chance to review everything on your list.

Some comments:

>  Conditional fields implementation using literal_eval (security risk?)

We only evaluate the widget's attributes defined within code, never anything that can be manipulated client side.

> No escaping for field labels, help text, attributes, values (XSS vector)

Could you elaborate this a little for me, please? Is this referring to forms created within Wagtail and assuming a malicious user has gained access? Defining markup within help text, for example, is entirely plausible.

**Comment by thibaudcolas** [comment]
@kbayliss 🙌  all the changes here (aside from the auto-formatting) are tooling only, so I’d recommend you review/merge this PR as-is – I won’t be looking at that side of things any further.

For the list above, I’ll make PRs either here, or in our closed-source integration code, or just comments in a new issue if I’m not sure what to suggest. I’m planning to go through this tomorrow, worst case next week.

**Review by kbayliss — APPROVED** [review]
This is looking great, thanks @thibaudcolas! I re-added the LGTM badge as I couldn't see why that would be removed? Let me know if I should remove it though...

**Comment by thibaudcolas** [comment]
🙌 

In the case of LGTM, this is because I’ve set up GitHub’s own CodeQL instead. Configuration: https://github.com/torchbox/tbxforms/blob/main/.github/workflows/codeql.yml. Results: https://github.com/torchbox/tbxforms/security/code-scanning

GitHub owns Semmle, the company who is behind LGTM (and built CodeQL), so I’d expect LGTM.com will eventually go away.

**Pull request merged**
